scipy/optimize/_bracket.py

```diff
@@ -199,32 +211,32 @@ def _bracket_root(func, xl0, xr0=None, *, xmin=None, xmax=None, factor=None,
     # bracket `x0` and the moving end `x` will grow by `factor` each iteration.
     # For searches with a limit, the distance between the `limit` and moving
     # end of the bracket `x` will shrink by `factor` each iteration.
-    i = np.isinf(limit)
+    i = xp.isinf(limit)
     ni = ~i
-    d = np.zeros_like(x)
+    d = xp.zeros_like(x)
     d[i] = x[i] - x0[i]
     d[ni] = limit[ni] - x[ni]
 
-    status = np.full_like(x, eim._EINPROGRESS, dtype=int)  # in progress
+    status = xp.full_like(x, eim._EINPROGRESS, dtype=xp.int32)  # in progress
```

Comment:
Yup, `int32` is what I've been going with for all the integer outputs. We want negative numbers for `status` sometimes, and the function evaluation count can get pretty high, so `int32` seems OK. I suppose we run the risk of it overflowing if there are a few billion function evaluations, but that should never happen in a bracketing function.